test/GroupsTests.jl

```diff
@@ -43,6 +43,10 @@ gtrial = BenchmarkGroup([], Dict("t" => trial))
 #-------#
 
 @test BenchmarkGroup() == BenchmarkGroup([], Dict())
+@test BenchmarkGroup([]) == BenchmarkGroup([], Dict())
+@test typeof(BenchmarkGroup([], Dict{Int,Int}(1=>2)).data) == Dict{Any,Any}
+@test BenchmarkGroup([], Dict(1=>2)).data == Dict(1=>2)
+@test BenchmarkGroup([], Base.ImmutableDict(1=>2)).data == Base.ImmutableDict(1=>2)
```

Comment:
This is  a very simple thing storing the Associative you wanted to store in the right field, I think doing it with OrderedDict is an overkill, but okay I can add that dependency.